gslib/gcs_json_api.py

```diff
@@ -251,8 +252,13 @@ def __init__(self,
     else:
       self.authorized_download_http = self.download_http
       self.authorized_upload_http = self.upload_http
-    WrapDownloadHttpRequest(self.authorized_download_http)
-    WrapUploadHttpRequest(self.authorized_upload_http)
+
+    if isinstance(self.credentials, P12Credentials):
```

Comment:
This check is used in other places as well? Can we turn it into a function so we can re-use?